### PR TITLE
use 2 unit test splits for osx, to reduce concurrency

### PR DESCRIPTION
**Pull request by dholth**
### Description

Two unit-tests splits for OSX. Will save us ~6 minutes of setup time since we avoid running the third split. Since the integration tests are much slower, a bigger unit tests run won't hold anything up.

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -283,35 +283,28 @@ jobs:
...
             test-group: '3'
+          - test-type: unit
+            test-splits: 3
+          - test-type: integration
+            test-splits: 2
+          # odd choice kept for parity with previous commit
+          - test-type: integration
+            default-channel: 'conda-forge'
+          - test-type: unit
+            default-channel: 'defaults'
```
Comment (kenodegard) [inline .github/workflows/tests.yml:303]: Isn't this oddity actually a bug?

File: tests/test_resolve.py [hunk 2]
```diff
@@ -127,7 +127,7 @@ def test_anaconda_nomkl(self):
 
 def test_generate_eq_1():
     # avoid cache from other tests which may have different result
-    r._reduced_index_cache = {}
+    _index, r = get_index_r_1()
```
Comment (kenodegard) [inline tests/test_resolve.py:130]: Doesn't belong in this PR

Comment (dholth) [reply tests/test_resolve.py:130]: If I leave this out of this PR, then the tests fail.

Comment (kenodegard) [reply tests/test_resolve.py:130]: that's odd, shouldn't this be failing on `main` too then?

Comment (dholth) [reply tests/test_resolve.py:130]: It is re-using a module-scope cached `r`, and something (no idea what) in another test sets it up for success or failure before we run here. So these new splits have improved the quality of our test suite.

The `least_duration` algorithm isn't as aggressive as randomizing test order, but it can split each file's tests into multiple runners compared to the naive algorithm which runs the 1st third of the tests in one batch, and the 2nd third in another batch and so on.

Comment (kenodegard) [reply tests/test_resolve.py:130]: ah I see, carry on then

Comment (dholth) [reply tests/test_resolve.py:130]: auto-merge enabled

File: .github/workflows/tests.yml (hunk 1 above)
Comment (dholth) [reply .github/workflows/tests.yml:303]: If we must change this to something besides "only run on defaults" or "randomly pick which channel to run against", let's wait until we have enough runners.

**Comment by dholth** [comment]
I don't have anything else to add to this PR, ready to merge.

**Review by kenodegard — APPROVED** [review]
